src/linalg.cpp

```diff
@@ -0,0 +1,319 @@
+#include "linalg.h"
+
+/**
+ * @brief Solves a generalized eigenvalue problem using the Power method.
+ * @details This function takes in a loss + streaming Matrix (A),
+ *          a fission gain Matrix (M), a flux Vector (X), a tolerance used
+ *          for both the power method and linear solve convergence (tol), and
+ *          a successive over-relaxation factor (SOR_factor) and computes the
+ *          dominant eigenvalue and eigenvector using the Power method. The
+ *          eigenvalue is returned along and the input X Vector is modified in
+ *          place to be the corresponding eigenvector.
+ * @param A the loss + streaming Matrix object
+ * @param M the fission gain Matrix object
+ * @param X the flux Vector object
+ * @param tol the power method and linear solve source convergence threshold
+ * @param SOR_factor the successive over-relaxation factor
+ * @return k_eff the dominant eigenvalue
+ */
+FP_PRECISION eigenvalueSolve(Matrix* A, Matrix* M, Vector* X, FP_PRECISION tol,
+                             FP_PRECISION SOR_factor) {
+
+  log_printf(INFO, "computing eigenvalue...");
```

Comment:
Make this log msg more explicit and informative - something brief but it should be clear that this is not coming from the MOC solver